kasa/discover.py

```diff
@@ -423,7 +423,7 @@ class EncryptionScheme(BaseModel):
     mac: str
     mgt_encrypt_schm: EncryptionScheme
 
-    device_id: Optional[str] = Field(default=None, alias="device_type_hash")
+    device_id: Optional[str] = Field(default=None, alias="device_id_hash")
```

Comment:
This changes the library code, was it broken already or ..?

Reply:
Yes I know. It seems I gave it a bad alias the first time round but I don’t think this will really affect anyone.

Reply:
So the alias is a the key for json (de)serialization, defining it is not mandatory but allows using nicer property names for some abbreviated ones found in the payloads. That's why I suggested using one for the `mgt_encrypt_schm`, but that didn't work, right?

Reply:
The alias is used by the library when it serializes it to a dict for basic display via `update_from_discover_info()` (i.e. there's no `alias` field so we alias IP to `alias`) .  It's not currently used the other way round, i.e. for de-serialization.  It might work the other way round but I think it would be confusing having a mix of both.

N.B. This did make me realise that I shouldn't have serialised the aliases for the test fixtures so I've done the following in `dump_devinfo` to ensure the fixture contains what we actually get from the device:

```
    if device._discovery_info:
        # Need to recreate a DiscoverResult here because we don't want the aliases
        # in the fixture, we want the actual field names as returned by the device.
        dr = DiscoveryResult(**device._discovery_info)
        final["discovery_result"] = dr.dict(
            by_alias=False, exclude_unset=True, exclude_none=True, exclude_defaults=True
        )
```

Reply:
Ah, okay. I'm not sure if this is the best solution, but this is just a devtool so I hink it's fine :+1: